jenkins/ompi/ompi_test.sh

```diff
@@ -377,549 +331,566 @@ function oshmem_runner()
-                    if [ $val -gt 0 ]; then
-                        echo "Test for the dist mapping policy was incorrectly launched or BIOS doesn't provide necessary information."
-                    else
-                        exit 1
-                    fi
-                fi
-            done
-        fi
-    fi
-    set -e
-}
-
-
 trap "on_exit" INT TERM ILL FPE SEGV ALRM
 
 on_start
 
+if [ "${ci_test_build}" = "yes" ]
+then
+    echo "Building Open MPI..."
 
-
-if [ "$jenkins_test_build" = "yes" ]; then
-    echo "Building OMPI"
-
-    if [ -x "autogen.sh" ]; then
-        autogen_script=./autogen.sh
+    if [ -x "autogen.sh" ]
+    then
+        autogen_script="./autogen.sh"
     else
-        autogen_script=./autogen.pl
+        autogen_script="./autogen.pl"
     fi
 
     # control mellanox platform file, select various configure flags
-    export mellanox_autodetect=yes
-    export mellanox_debug=yes
+    export mellanox_autodetect="yes"
+    export mellanox_debug="yes"
 
-    configure_args="--with-platform=contrib/platform/mellanox/optimized --with-ompi-param-check --enable-picky $extra_conf"
+    configure_args="--with-platform=contrib/platform/mellanox/optimized --with-ompi-param-check --enable-picky ${extra_conf}"
 
-    if [ "$jenkins_test_ucx" = "yes" ]; then
+    if [ "${ci_test_ucx}" = "yes" ]
+    then
         module load hpcx-gcc-stack
-        if [ "$jenkins_test_use_ucx_branch" = "yes" ]; then
-            export ucx_root=$WORKSPACE/ucx_local
-            git clone https://github.com/openucx/ucx -b $jenkins_test_ucx_branch $ucx_root
-            (cd $ucx_root;\
-            ./autogen.sh;\
-            ./contrib/configure-release --prefix=$ucx_root/install;\
-            make -j9 install; )
+
+        if [ "${ci_test_use_ucx_branch}" = "yes" ]
+        then
+            export ucx_root="$WORKSPACE/ucx_local"
+            git clone https://github.com/openucx/ucx -b ${ci_test_ucx_branch} "${ucx_root}"
+            (cd "${ucx_root}";\
+                ./autogen.sh;\
+                ./contrib/configure-release --prefix="${ucx_root}/install";\
+                make -j"$(nproc)" install; )
            export UCX_DIR=$ucx_root/install
 
            # We need to override LD_LIBRARY_PATH because.
            # `module load hpcx-gcc-stack` will pull the legacy
            # UCX files that will interfere with our custom-built
            # UCX during configuration and the runtime I guess
-           export LD_LIBRARY_PATH=${HPCX_UCX_DIR}/lib:$LD_LIBRARY_PATH
+           export LD_LIBRARY_PATH="${HPCX_UCX_DIR}/lib:${LD_LIBRARY_PATH}"
         fi
-        export ucx_dir=$HPCX_UCX_DIR
+
+        export ucx_dir=${HPCX_UCX_DIR}
     fi
 
-    rm -rf $ompi_home_list
+    rm -rf "${ompi_home_list}"
 
     # build ompi
-    $autogen_script
-    echo ./configure $configure_args --prefix=$OMPI_HOME1 | bash -xeE
-    make $make_opt install
-    jenkins_build_passed=1
+    ${autogen_script}
```

Comment:
If you're going to use parallelism with `make` (i.e., `make -j ...`), you should probably also set the env variable `AUTOMAKE_JOBS` to the same `-j` value that you pass to `make`.  This can significantly speed up the time required to run `autogen`.

Reply:
Fixed.